tests/foreman/api/test_contentview.py

```diff
@@ -1256,6 +1256,54 @@ def test_positive_publish_multiple_repos(self):
             content_view.publish()
         self.assertEqual(len(content_view.read().version), 1)
 
+    @tier2
+    def test_composite_content_view_with_same_repos(self):
+        """Create a Composite Content View with content views having same yum repo.
+        Add filter on the content views and check the package count for composite content view
+        should not be changed.
+
+        :id: 957f3758-ca1e-4a1f-8e7d-171750e0eb87
+
+        :expectedresults: package count for composite content view should not be changed in
+            case of mismatch.
+
+        :bz: 1639390
+
+        :CaseImportance: Medium
+        """
+        org = self.org
+        product = self.product
+        repo = entities.Repository(
+            content_type='yum',
+            product=product,
+            url=CUSTOM_MODULE_STREAM_REPO_2).create()
+        repo.sync()
+        content_view_1 = entities.ContentView(organization=org).create()
+        content_view_2 = entities.ContentView(organization=org).create()
+
+        # create content views with same repo and different filter
+        for content_view, package in [(content_view_1, 'camel'), (content_view_2, 'cow')]:
+            content_view.repository = [repo]
+            content_view.update(['repository'])
+            content_view_info = apply_package_filter(content_view, repo,
+                                                     package, inclusion=False,)
+            assert content_view_info.package_count == 35
+
+        # create composite content view with these two published content views
+        comp_content_view = entities.ContentView(
+            composite=True,
+            organization=org,
+        ).create()
+        content_view_1 = content_view_1.read()
+        content_view_2 = content_view_2.read()
+        for content_view_version in [content_view_1.version[-1], content_view_2.version[-1]]:
+            comp_content_view.component.append(content_view_version)
+            comp_content_view = comp_content_view.update(['component'])
+        comp_content_view.publish()
+        comp_content_view = comp_content_view.read()
+        comp_content_view_info = comp_content_view.version[0].read()
+        assert comp_content_view_info.package_count == 36
```

Comment:
Can you add a check or two that some expected packages are included, and another that excluded packages aren't present?

Reply:
Here we include or exclude the rpms from content views attached to composite_content_view which are build by same yum repo will create a mismatch. In this special case, we are making a behaviour such that not filters should get applied. You may have missed this comment https://github.com/SatelliteQE/robottelo/pull/7423#discussion_r337448064

Although adding one more layer of assertion it will just increase the time of execution. If you think if that still valid in this case I will add it.

Reply:
@JacobCallahan I have added the one more layer of assertion based on your comments. Please take a look. 

https://github.com/SatelliteQE/robottelo/pull/7423/commits/3fdce5949126a84296a564004bc4b7abf9fede74

Reply:
Thanks, @omkarkhatavkar! The extra time for these assertions shouldn't amount to more than a couple of ms, but the checks themselves are valuable :)